mm2src/coins/eth.rs

```diff
@@ -5015,31 +5011,37 @@ fn checksum_address(addr: &str) -> String {
 /// The input must be 0x prefixed hex string
 fn is_valid_checksum_addr(addr: &str) -> bool { addr == checksum_address(addr) }
 
-/// Requests the nonce from all available nodes and checks that returned results equal.
-/// Nodes might need some time to sync and there can be other coins that use same nodes in different order.
-/// We need to be sure that nonce is updated on all of them before and after transaction is sent.
+/// Requests the nonce from all available nodes and returns the highest nonce available with the list of nodes that returned the highest nonce.
+/// Transactions will be sent using the nodes that returned the highest nonce.
 #[cfg_attr(test, mockable)]
-fn get_addr_nonce(addr: Address, web3s: Vec<Web3Instance>) -> Box<dyn Future<Item = U256, Error = String> + Send> {
+fn get_addr_nonce(
+    addr: Address,
+    web3s: Vec<Web3Instance>,
+) -> Box<dyn Future<Item = (U256, Vec<Web3Instance>), Error = String> + Send> {
     let fut = async move {
         let mut errors: u32 = 0;
         loop {
```

Comment:
You can change this with a repeatable future since we started to work on this function

Reply:
Sure, it can be done. Will have to change the function to async to do that.

Reply:
Unfortunately, repeatable future can't be used in the above case because of this
https://github.com/KomodoPlatform/atomicDEX-API/blob/ee370707c68e42596ee084da97eaa43384a8f42c/mm2src/common/custom_futures/repeatable.rs#L3-L21 since errors counter should be used instead of counter https://github.com/KomodoPlatform/atomicDEX-API/blob/c5047d40e66a1c287f20a50e9d58e711ab3f3740/mm2src/coins/eth.rs#L5022

Reply:
I didn't know about this, then it can stay as it is